libs/x-content/src/main/java/org/elasticsearch/common/xcontent/DeprecationHandler.java

```diff
@@ -97,4 +97,13 @@ public void usedDeprecatedField(String parserName, Supplier<XContentLocation> lo
      */
     void usedDeprecatedField(String parserName, Supplier<XContentLocation> location, String usedName);
 
+    /**
+     * Returns a variant of the deprecation handler that emits compatible logs or deprecation logs
+     * if a deprecation handler does not support compatible logging then returns the same instance.
+     * @param compatibleWarnings a flag indicating which handler to return
+     * @return a variant of the current deprecation handler
+     */
+    default DeprecationHandler getInstance(boolean compatibleWarnings) {
+        return this;
```

Comment:
This seems off ... such that I can pass true or false and get back the same instance. 

I would suggest to not use a default method if there are variants possible.

Reply:
I agree, that this being part of an interface is a design flaw. 
I think there is a way to choose a LoggingDeprecationHandler when creating an xcontent parser.
The RestRequest.contentParser() is aware of the version and can choose which one to use.

Reply:
actually I take that back (the restrequest part) I cannot choose the DeprecationHandler at that point, because compatible request can still access V_8 (non compatible fields, but newly deprecated) and those should not emit compatible log..